### ISSUE TITLE
VS2017RC4: Opening Package Manager Console results in error

### ISSUE THREAD

**Issue by rrelyea** [NuGet/Home#4564]
internal bug 382548

Repro Steps:Via extensions, I selected an extension that required closing VS2017. 
Extension isntalled with no problems. This extension has worked with previous builds of VS2017 RC. 
Re-Open VS2017RC4 
Open existing solution 
Open package manager console 
I'm gettig an error message in the console that says "The operation failed as details for project MovieList could not be loaded." 
The error is in red text. It follows the standard PMC start up text 
Version: Package Manager Console Host Version 4.0.0.2275 
There is no prompt and all of the PMC toolbar items are disabled. 
The solution builds with no problems. 
I closed and reopened VS without opening a soluiton and the PMC opened just fine. 
I closed teh PMC and opened my solution. Then opened the PMC again and it was okay. 
I closed the PMC. Closed VS2017. Opened VS2017. Opened the solution. Then opened the PMC. It opened correctly and gave me the prompt. 
So I think it may have something to do with installing a VSIX. I don't think it is the particular extension though (SQL CE SQLite toolbox) 

Impact:a) confusion 
b) had to close and reopen Visual Studio

**Comment by alpaix** [comment]
Was able to repro with the most recent bits in dev. Not related to extension installed. The key step is restarting a VS and opening PMC when solution is loading. 

The error entry from the activity log:
```xml
<entry>
    <record>760</record>
    <time>2017/02/11 01:00:58.758</time>
    <type>Error</type>
    <source>NuGet Package Manager</source>
    <description>System.InvalidOperationException: The operation failed as details for project WebApplication3 could not be loaded.&#x000D;&#x000A;   at NuGet.PackageManagement.VisualStudio.CpsPackageReferenceProject.GetAssetsFilePathAsync()&#x000D;&#x000A;   at NuGet.ProjectManagement.BuildIntegratedProjectUtility.&lt;GetLockFileOrNull&gt;d__1.MoveNext()&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()&#x000D;&#x000A;   at NuGet.ProjectManagement.BuildIntegratedProjectUtility.&lt;GetOrderedProjectPackageDependencies&gt;d__0.MoveNext()&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()&#x000D;&#x000A;   at NuGetConsole.Host.PowerShell.Implementation.PowerShellHost.&lt;ExecuteInitScriptsAsync&gt;d__66.MoveNext()&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()&#x000D;&#x000A;   at NuGetConsole.Host.PowerShell.Implementation.PowerShellHost.&lt;&gt;c__DisplayClass63_0.&lt;&lt;Initialize&gt;b__0&gt;d.MoveNext()</description>
  </entry>
```

**Comment by rrelyea** [comment]
May be related to your recent change? Can you please look.

**Comment by joelverhagen** [comment]
Yup, definitely looks related. The key clue here is the `ExecuteInitScriptsAsync` in the stack trace. I'll look into this on Monday.

**Comment by joelverhagen** [comment]
Formatted stack:
```
System.InvalidOperationException: The operation failed as details for project WebApplication1 could not be loaded.
   at NuGet.PackageManagement.VisualStudio.CpsPackageReferenceProject.GetPackageSpecsAsync(DependencyGraphCacheContext context)
   at NuGet.PackageManagement.DependencyGraphRestoreUtility.<GetSolutionRestoreSpec>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.DependencyGraphRestoreUtility.<IsRestoreRequiredAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.SolutionRestoreManager.SolutionRestoreJob.<RestorePackageSpecProjectsAsync>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at NuGet.SolutionRestoreManager.SolutionRestoreJob.<RestoreAsync>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at NuGet.SolutionRestoreManager.SolutionRestoreJob.<RestoreAsync>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.SolutionRestoreManager.SolutionRestoreJob.<ExecuteAsync>d__15.MoveNext()
```

I was able to repro it by trying to execute a PMC command (`1 + 1`, <kbd>Enter</kbd>).

The stack is different... will look into both of these issues...

**Comment by alpaix** [comment]
@joelverhagen That's different stack and somewhat expected if not all of solution projects loaded yet. What do you see in the output window?

**Comment by joelverhagen** [comment]
```
> 1 + 1
(cursor is hanging here -- no message)
```

**Comment by alpaix** [comment]
I mean Package Manager output (not the PMC).

**Comment by joelverhagen** [comment]
This fix is address by PR https://github.com/NuGet/NuGet.Client/pull/1186, which is signed off by @emgarten, @alpaix, and @jainaashish.

**Comment by rrelyea** [comment]
It appears that RC2, 3, and 4 had this problem.
It appears very hard to hit. So post RTM.

**Comment by mishra14** [comment]
Seems like: https://github.com/NuGet/Home/issues/4659

**Comment by joelverhagen** [comment]
This fix has been checked into `release-4.1.0-rtm` as well as `dev` (for 4.3.0).
